cai_causal_graph/causal_graph.py

```diff
@@ -405,11 +405,13 @@ def __iter__(self) -> Iterator:
         for k, v in self.to_dict().items():
             yield k, v
 
-    def __eq__(self, other: object) -> bool:
+    def __eq__(self, other: object, deep: bool = False) -> bool:
         """
         Check if equal to another `cai_causal_graph.causal_graph.CausalGraph`.
 
         Checks if all nodes and edges are equal.
+        :param other: The other `cai_causal_graph.causal_graph.CausalGraph` to compare to.
+        :param deep: If `True`, also check if all the attributes of nodes and edges are equal. Default is `False`.
```

Comment:
Not all attributes as we know some are still ignored (e.g., incoming and outgoing edges on nodes). Would say "If `True`, also does deep equality checks on all the nodes and edges."